oak_remote_attestation/proto/v1/evidence.proto

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.remote_attestation.v1;
+
+option java_multiple_files = true;
+option java_package = "com.google.oak.remote_attestation.v1";
+
+// This proto defines the layered DICE Attestation Evidence.
+//
+// DICE provides a mechanism for combining software measurements and corresponding certificates into
+// a chain, where each element (called “layer”) represents a piece of software loaded into the VMs
+// memory. These layers are loaded sequentially, i.e. the previous layer loads the next layer. The
+// previous layer is also responsible for measuring the next layer, generating a private key and a
+// certificate for it.
+//
+// <https://trustedcomputinggroup.org/wp-content/uploads/TCG_DICE_Attestation_Architecture_r22_02dec2020.pdf>
+// <https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf>
+
+enum TeePlatform {
+  UNSPECIFIED = 0;
+  AMD_SEV_SNP = 1;
+  INTEL_TDX = 2;
+}
+
+message RootLayerEvidence {
+  // The platform providing the attestation report.
+  TeePlatform platform = 0;
+  // TEE-specific attestation report acting as a non-standard certificate for the
+  // Layer0 ECA public key.
+  bytes remote_attestation_report = 1;
+  // Serialised ECA public key for Layer0.
+  bytes eca_public_key = 2;
+}
+
+message LayerEvidence {
+  string layer_name = 0;
+  // CBOR/COSE/CWT ECA certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  bytes eca_certificate = 1;
+}
+
+message ApplicationKeys {
```

Comment:
Doc comment? 
What are these and where do they come from?
How are these related to the other evidence, given that this is included in `Evidence` next to `RootLayerEvidence` and `LayerEvidence`. 

Should this be renamed to `FinalLayerEvidence`?

Reply:
I also was expecting this to be called something about Layer Evidence

Reply:
This is not an evidence, it's just a proto containing signing and encryption keys.

Reply:
While I initially proposed this name in the doc, I agree with Tiziano here. The certificates will contain measurements, so will also act as evidence.

Reply:
I'm not sure we should call it `ApplicationLayerEvidence` because we will have a separate DICE `LayerEvidence` for the last layer.

Even `ApplicationEvidence` may be confusing.